src/auth/awsclient.ts

```diff
@@ -197,10 +218,21 @@ export class AwsClient extends BaseExternalAccountClient {
     );
   }
 
+  private async getAwsSessionToken(): Promise<string> {
+    const opts: GaxiosOptions = {
+      url: this.awsSessionTokenUrl,
+      method: 'PUT',
+      responseType: 'text',
+      headers: {'x-aws-ec2-metadata-token-ttl-seconds': '21600'},
+    };
+    const response = await this.transporter.request<string>(opts);
+    return response.data;
+  }
+
   /**
    * @return A promise that resolves with the current AWS region.
    */
-  private async getAwsRegion(): Promise<string> {
+  private async getAwsRegion(headers: Headers): Promise<string> {
```

Comment:
Suggest adding the new headers field in the javadoc comments here and elsewhere for consistency.